src/Microsoft.AspNet.Diagnostics/StatusCodePagesMiddleware.cs

```diff
@@ -25,8 +25,17 @@ public StatusCodePagesMiddleware(RequestDelegate next, [NotNull]StatusCodePagesO
 
         public async Task Invoke(HttpContext context)
         {
+            context.SetFeature<IStatusCodeFeature>(new StatusCodeFeature());
+
             await _next(context);
 
+            var statusCodeFeature = context.GetFeature<IStatusCodeFeature>();
+            if (statusCodeFeature != null && !statusCodeFeature.Enabled)
```

Comment:
>Is there any benefit to knowing that the application is capable of this feature, but that it is currently disabled?

Currently nothing unless we add more to the feature interface itself which will require such a check.